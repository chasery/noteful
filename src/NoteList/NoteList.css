.NoteList {
  flex: 3;
}

.NoteList > ul {
  margin: 0;
  padding: 0;
}
