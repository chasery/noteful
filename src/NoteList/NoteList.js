import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import NotefulContext from "../NotefulContext";
import Note from "../Note/Note";
import "./NoteList.css";

class NoteList extends Component {
  static contextType = NotefulContext;
  static defaultProps = {
    notes: [],
  };

  render() {
    const notes = this.context.notes
      .filter(
        (note) =>
          !this.props.match.params.folderId ||
          note.folderId === this.props.match.params.folderId

        // if (this.props.match.params.folderId) {
        //   return note.folderId === this.props.match.params.folderId;
        // } else {
        //   return note;
        // }
      )
      .map((note) => {
        return (
          <Note
            key={note.id}
            id={note.id}
            name={note.name}
            modified={note.modified}
            folderId={note.folderId}
          />
        );
      });

    return (
      <section className="NoteList">
        <ul>{notes}</ul>
        <button disabled>Add Note</button>
      </section>
    );
  }
}

export default withRouter(NoteList);
