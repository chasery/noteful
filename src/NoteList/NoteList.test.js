import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NoteList from "./NoteList";

describe("NoteList component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<NoteList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
