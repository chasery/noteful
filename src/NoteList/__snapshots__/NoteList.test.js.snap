// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList component renders without error 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
